tlsproxy.go

```diff
@@ -128,18 +139,20 @@ func doRun(listen func() (net.Listener, error), dial func() (net.Conn, error)) {
 	log.Debugf("Listening for incoming connections at: %v", l.Addr())
 
 	for {
-		in, err := l.Accept()
+		_in, err := l.Accept()
 		if err != nil {
 			log.Fatalf("Unable to accept: %v", err)
 		}
 
+		in := idletiming.Conn(_in, *idleTimeout, nil)
```

Comment:
idletiming creates a goroutine and a timer, which may have some performance penalty I guess? Do you think it's useful to benchmark it against a version that using `SetKeepAlive` and `SetKeepAlivePeriod`?